simpl/threadlocal.py

```diff
@@ -0,0 +1,52 @@
+"""Threadlocal."""
+import collections
+import threading
+
+THREAD_STORE = threading.local()
+DEFAULT_NAMESPACE = 'call_context'
+
+
+class ThreadLocalDict(collections.MutableMapping):
+
+    """A dict whose data is local to the thread."""
+
+    def __init__(self, varname, *args, **kwargs):
+        self.varname = varname
+        self.args = args
+        self.kwargs = kwargs
+
+    def __repr__(self):
+
+        under = repr(self._get_local_dict())
+        return '<ThreadLocalDict %s>' % under
+
+    def _get_local_dict(self):
+        """Retrieve (or initialize) the thread-local data to use."""
+        local_var = getattr(THREAD_STORE, self.varname, None)
+        if not local_var:
+            local_var = dict(*self.args, **self.kwargs)
+            setattr(THREAD_STORE, self.varname, local_var)
+        return local_var
```

Comment:
If we use the presence of the attr as the test, then the thread local dict only gets initialized once (and can be falsy):
```python
    def _get_local_dict(self):
        """Retrieve (or initialize) the thread-local data to use."""
        try:
            return getattr(THREAD_STORE, self.varname)
        except AttributeError:
            local_var = dict(*self.args, **self.kwargs)
            setattr(THREAD_STORE, self.varname, local_var)
            return local_var
```